### PR TITLE
circleci: Run sanitizers on a larger system

### PR DESCRIPTION
**Pull request by dridi**
This is an attempt at speeding up this job that usually takes three time as long as other distcheck jobs.

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -433,8 +438,9 @@ workflows:
           name: build_ubuntu_noble
           dist: ubuntu
           release: noble
+          rclass: large
           extra_conf: --enable-asan --enable-ubsan --enable-workspace-emulator
-          make_target: check -j2
+          make_target: check
```
Comment (asadsa92) [inline .circleci/config.yml:443]: Is this a typo? The large instance have 4 vCPU (instead of 2 for medium), should we not use `-j4` here?

Comment (dridi) [reply .circleci/config.yml:443]: `-j4` is the default, so we no longer need a `-j2` override.

**Review by asadsa92 — APPROVED** [review]
LGTM, we reduced the runtime of ASAN by 50% but I think we can do better (see my comment).

I inspected at our usage and we have only been using maximum 44% of the credits we have been given by CCI for open-source. So, it should be fine in that area,

**Comment by dridi** [comment]
Approximately 15m (29m before this change) for the sanitizer job vs 11m for other jobs.

The large instance will consume more cci credits, but we should also save credits on time spent running jobs.

File: .circleci/config.yml (hunk 1 above)
Comment (asadsa92) [reply .circleci/config.yml:443]: Aha, nvm then.

**Pull request merged**